projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/graphql/ExecutionDataFetcher.java

```diff
@@ -114,14 +114,16 @@
    }
 
    private Page<DataJobExecution> findAllExecutions(DataJobExecutionQueryVariables dataJobExecutionQueryVariables, String dataJobName) {
-      DataJobExecutionFilter filter = dataJobExecutionQueryVariables.getFilter();
-      if (dataJobName != null && filter != null) {
-         if (filter.getJobNameIn() != null) {
-            throw new GraphQLException("The jobNameIn filter is not supported for nested executions");
-         }
-         filter = filter.toBuilder()
-                 .jobNameIn(List.of(dataJobName))
-                 .build();
+      DataJobExecutionFilter filter = dataJobExecutionQueryVariables.getFilter() != null ?
+            dataJobExecutionQueryVariables.getFilter() :
+            DataJobExecutionFilter.builder().build();
+
+      if (filter.getJobNameIn() != null) {
```

Comment:
Shouldn't we still check if `dataJobName` is not null here? Otherwise, the exception will be raised when getting all executions with a filter containing `jobNameIn`.

Reply:
Good catch :)